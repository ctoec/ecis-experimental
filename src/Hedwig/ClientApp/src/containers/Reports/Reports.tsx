import { useParams, useLocation } from "react-router";
import React from "react";
import ReportsSummary from './ReportsSummary/ReportsSummary';
import ReportDetail from './ReportDetail/ReportDetail';
import { useAlertContext, AlertProvider } from "../../contexts/Alert/AlertContext";
import { AlertProps } from "../../components/Alert/Alert";


export default function Reports() {
	const { id } = useParams();
	const { state: alerts } = useLocation<AlertProps[]>();
	const alertContext = useAlertContext(alerts);
	
	let reportContainer;
	if (!id) {
		reportContainer = <ReportsSummary></ReportsSummary>;
	} else {
		reportContainer = <ReportDetail></ReportDetail>;
	}

	return (
		<AlertProvider value={alertContext}>
			{reportContainer}
		</AlertProvider>
	)
}