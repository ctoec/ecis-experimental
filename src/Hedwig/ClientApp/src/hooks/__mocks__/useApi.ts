const moment = require('moment');
import {
	Age,
	CdcReport,
	Enrollment,
	FundingSource,
	FundingTime,
	Gender,
	Region,
	ApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdGetRequest,
	ApiOrganizationsOrgIdEnrollmentsGetRequest,
} from '../../generated';

type ChangeField = { keys: string[]; newValue?: any };
const swapFields = <T>(inputObject: T, changeFields: ChangeField[]): T => {
	// Make a deep copy to avoid changing the original
	const newObject = JSON.parse(JSON.stringify(inputObject));
	changeFields.forEach(field => {
		let changeObject = newObject;
		field.keys.forEach((key, i) => {
			// Accessors are an ordered array of strings referring to nested keys
			if (i === field.keys.length - 1) {
				// If it's the last key in the array, then it's the thing that needs to be changed to a new value
				changeObject[key] = field.newValue;
			} else {
				// Otherwise keep digging
				changeObject = changeObject[key];
			}
		});
	});
	return newObject;
};

const reportEnrollmentValidationError = [
	{
		message: 'Enrollments have validation errors',
		isSubObjectValidation: true,
		field: 'Enrollments',
	},
];

const enrollmentValidationError = [
	{
		message: 'Child has validation errors',
		isSubObjectValidation: true,
		field: 'Child',
	},
];

export const completeEnrollment: Enrollment = {
	id: 1,
	childId: '2',
	siteId: 1,
	ageGroup: Age.Preschool,
	entry: new Date('2018-02-03'),
	exit: null,
	child: {
		id: '2',
		firstName: 'Lily',
		middleName: 'Luna',
		lastName: 'Potter',
		birthdate: new Date('2016-12-12'),
		birthTown: 'Hogsmeade',
		birthState: 'CT',
		birthCertificateId: '123',
		nativeHawaiianOrPacificIslander: true,
		hispanicOrLatinxEthnicity: true,
		gender: Gender.Female,
		foster: false,
		familyId: 1,
		organizationId: 1,
		family: {
			id: 1,
			addressLine1: '4 Privet Drive',
			town: 'Hogsmeade',
			state: 'CT',
			zip: '77777',
			homelessness: false,
			organizationId: 1,
			determinations: [{ id: 1, notDisclosed: true, familyId: 1 }],
		},
	},
	fundings: [
		{
			id: 1,
			enrollmentId: 1,
			source: FundingSource.CDC,
			familyId: 1,
			certificateStartDate: new Date('2019-03-01'),
			firstReportingPeriodId: 1,
			firstReportingPeriod: {
				id: 1,
				type: FundingSource.CDC,
				period: new Date('2019-03-01'),
				periodStart: new Date('2019-03-01'),
				periodEnd: new Date('2019-03-31'),
				dueAt: new Date('2019-04-15'),
			},
			time: FundingTime.Full,
		},
	],
};

export const enrollmentMissingBirthCertId = swapFields(completeEnrollment, [
	{ keys: ['child', 'birthCertificateId'], newValue: undefined },
	{ keys: ['id'], newValue: 2 },
	{ keys: ['validationErrors'], newValue: enrollmentValidationError },
]);

export const enrollmentMissingFirstName = swapFields(completeEnrollment, [
	{ keys: ['child', 'firstName'], newValue: undefined },
	{ keys: ['id'], newValue: 3 },
	{ keys: ['validationErrors'], newValue: enrollmentValidationError },
]);

export const enrollmentMissingAddress = swapFields(completeEnrollment, [
	{ keys: ['child', 'family', 'addressLine1'], newValue: undefined },
	{ keys: ['id'], newValue: 4 },
	{ keys: ['validationErrors'], newValue: enrollmentValidationError },

	{
		keys: ['child', 'family', 'validationErrors'],
		newValue: [
			{
				message: 'Street address is required',
				isSubObjectValidation: false,
				field: 'AddressLine1',
			},
		],
	},
]);

export const enrollmentWithLaterStart = swapFields(completeEnrollment, [
	{ keys: ['entry'], newValue: new Date('2019-03-01') },
	{ keys: ['id'], newValue: 5 },
]);

export const allFakeEnrollments = [
	{
		enrollment: completeEnrollment,
	},
	{
		enrollment: enrollmentMissingBirthCertId,
	},
	{
		enrollment: enrollmentMissingFirstName,
		mutationError: {
			errors: { 'Child.FirstName': ['The FirstName field is required.'] },
			type: 'https://tools.ietf.org/html/rfc7231#section-6.5.1',
			title: 'One or more validation errors occurred.',
			status: 400,
		},
	},
	{
		enrollment: enrollmentMissingAddress,
	},
	{
		enrollment: enrollmentWithLaterStart,
	},
];

export const defaultOrganization = {
	id: 1,
	name: 'Test Organization',
	fundingSpaces: [
		{
			source: FundingSource.CDC,
			ageGroup: Age.Preschool,
			time: FundingTime.Full,
			capacity: 2,
			organizationId: 1,
		},
	],
	sites: [
		{
			name: 'Test Site',
			region: Region.East,
			titleI: false,
			organizationId: 1,
		},
	],
};

export const defaultReport: CdcReport = {
	id: 1,
	organizationId: 1,
	accredited: true,
	type: FundingSource.CDC,
	enrollments: allFakeEnrollments.map(e => e.enrollment),
	reportingPeriod: {
		id: 1,
		type: FundingSource.CDC,
		period: new Date('2019-09-01'),
		dueAt: new Date('2019-10-15'),
		periodStart: new Date('2019-09-01'),
		periodEnd: new Date('2019-09-28'),
	},
	organization: defaultOrganization,
	familyFeesRevenue: 1000,
	validationErrors: reportEnrollmentValidationError,
};

export const laterReport: CdcReport = swapFields(defaultReport, [
	{
		keys: ['reportingPeriod'],
		newValue: {
			id: 1,
			type: FundingSource.CDC,
			period: new Date('2019-10-01'),
			dueAt: new Date('2019-11-15'),
			periodStart: new Date('2019-10-01'),
			periodEnd: new Date('2019-10-31'),
		},
	},
	{ keys: ['id'], newValue: 2 },
]);

export const earlierReport: CdcReport = swapFields(defaultReport, [
	{
		keys: ['reportingPeriod'],
		newValue: {
			id: 1,
			type: FundingSource.CDC,
			period: new Date('2019-08-01'),
			dueAt: new Date('2019-09-15'),
			periodStart: new Date('2019-08-01'),
			periodEnd: new Date('2019-08-31'),
		},
	},
	{ keys: ['id'], newValue: 3 },
	{ keys: ['submittedAt'], newValue: new Date('2019-09-14') },
	{ keys: ['validationErrors'], newValue: undefined },
]);

export default (query: (api: any) => any) => {
	return query({
		apiOrganizationsOrgIdEnrollmentsGet: (
			params: ApiOrganizationsOrgIdEnrollmentsGetRequest
		) => {
			const enrollments = allFakeEnrollments
				.filter(e => (params.siteIds || []).includes(e.enrollment.siteId))
				.filter(({ enrollment: e }) => {
					return (
						(!e.entry ? true : moment(e.entry).isBefore(params.endDate)) &&
						(!e.exit ? true : moment(e.exit).isAfter(moment(params.startDate)))
					);
				});
			return [false, null, enrollments];
		},
		apiOrganizationsOrgIdSitesSiteIdEnrollmentsIdGet: (
			params: ApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdGetRequest
		) => {
			const thisEnrollment = allFakeEnrollments.find(e => e.enrollment.id === params.id);
			if (!thisEnrollment) return;
			const mutate = (_: any) => {
				return new Promise((resolve, reject) => {
					thisEnrollment.mutationError
						? reject(thisEnrollment.mutationError)
						: resolve(thisEnrollment.enrollment);
				});
			};
			return [false, null, thisEnrollment.enrollment, mutate];
		},
		apiOrganizationsOrgIdSitesIdGet: (params: any) => [
			false,
			null,
			{
				id: 1,
				name: "Children's Adventure Center",
				organizationId: 1,
				enrollments: undefined,
				organization: undefined,
			},
		],
		apiOrganizationsOrgIdSitesSiteIdEnrollmentsGet: (params: any) => [
			false,
			null,
			[enrollmentWithLaterStart, completeEnrollment].filter(e => {
				return (
					(!e.entry ? true : moment(e.entry).isBefore(params.endDate)) &&
					(!e.exit ? true : moment(e.exit).isAfter(moment(params.startDate)))
				);
			}),
		],
		apiOrganizationsOrgIdReportsGet: (params: any) => [
			false,
			null,
			[defaultReport, laterReport, earlierReport],
			(_: any) => {
				return new Promise((resolve, reject) => {
					resolve(defaultReport);
					reject({});
				});
			},
		],
		apiOrganizationsIdGet: (params: any) => [false, null, defaultOrganization],
		apiOrganizationsOrgIdReportsIdGet: (params: any) => [false, null, defaultReport],
	});
};
