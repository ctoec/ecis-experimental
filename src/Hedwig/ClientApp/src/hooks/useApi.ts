import { useContext, useEffect, useState, DependencyList } from 'react';
import { Configuration, HedwigApi } from '../OAS-generated';
import LoginContext from '../contexts/Login/LoginContext';
import getCurrentHost from '../utils/getCurrentHost';

export type Reducer<TData> = (data: TData, result: TData) => TData;
export type Query<TData> = (api: HedwigApi) => Promise<TData>
export type Mutate<TData> = (
	query: Query<TData>,
	reducer?: Reducer<TData | undefined>
) => Promise<TData | undefined>

interface ApiParamOpts<T> {
	defaultValue?: T,
	skip?: boolean,
	callback?: (_: T) => void
}

interface ApiState<T> {
	loading: boolean,
	error: string | null,
	data?: T,
	skip: boolean,
}

type ApiResult<TData> = [
	boolean,
	(string | null),
	(TData | undefined),
	Mutate<TData>
];

export default function useApi<TData>(
	query: (api: HedwigApi) => Promise<TData>,
	deps: DependencyList = [],
	opts?: ApiParamOpts<TData>
): ApiResult<TData> {
	const {
		defaultValue,
		skip,
		callback
	} = {
		defaultValue: undefined,
		skip: false,
		callback: undefined,
		...opts
	}

	const [state, setState] = useState<ApiState<TData>>({
		loading: true,
		error: null,
		skip,
		data: defaultValue,
	});
	const { loading, error, data } = state;

	const { accessToken, withFreshToken } = useContext(LoginContext);
	useEffect(() => {
		withFreshToken();
	});

	const api = accessToken
		? new HedwigApi(
				new Configuration({
					basePath: getCurrentHost(),
					apiKey: `Bearer ${accessToken}`,
				})
		  )
		: null;

	useEffect(() => {
		if (!api || skip) {
			setState({...state, loading: false});
			return;
		}

		query(api)
			.then((result) => {
				setState({ ...state, loading: false, error: null, data: result });
				if (callback) {
					callback(result);
				}
			})
			.catch((error) => setState({ ...state, loading: false, error: error.toString() }));
	// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [...deps, accessToken]);

	const mutate: Mutate<TData> = (
	  query,
	  reducer = (_, result) => result,
	) => {
		if (!api) {
			setState({...state, loading: false});
			return Promise.reject("No api!");
		}


		return query(api).then((result) => {
			setState({ ...state, data: reducer(data, result)})
			return result;
		});
	};

	return [loading, error, data, mutate];
}
