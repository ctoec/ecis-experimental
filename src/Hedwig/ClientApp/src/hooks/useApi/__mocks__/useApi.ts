// This file is @deprecated
// It should be replaced by only mocking constructApi and parseError
// See ChildInfo.tests.tsx for more information on how to do that.

import moment from 'moment';
import {
	Enrollment,
	ApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdGetRequest,
	ApiOrganizationsOrgIdEnrollmentsGetRequest,
	Organization,
	CdcReport,
	Site,
	Child,
	ApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdPutRequest,
} from '../../../generated';
import { ApiParamOpts } from '..';
import { useEffect } from 'react';

export const mockApiOrganizationsOrgIdEnrollmentsGet = (enrollments: Enrollment[]) => (
	params: ApiOrganizationsOrgIdEnrollmentsGetRequest
) => {
	const _enrollments = enrollments
		.filter((e) => !params.siteIds || params.siteIds.includes(e.siteId))
		.filter((e) => {
			return (
				(!e.entry ? true : moment(e.entry).isBefore(params.endDate)) &&
				(!e.exit ? true : moment(e.exit).isAfter(moment(params.startDate)))
			);
		});
	return { loading: false, error: null, data: _enrollments };
};

export const mockApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdGet = (enrollments: Enrollment[]) => (
	params: ApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdGetRequest
) => {
	const thisEnrollment = enrollments.find((e) => e.id === params.id);
	if (!thisEnrollment) return { loading: false, error: { status: '400' }, data: null };
	return { loading: false, error: null, data: thisEnrollment };
};

export const mockApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdPut = (enrollments: Enrollment[]) => (
	params: ApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdPutRequest
) => {
	const thisEnrollment = enrollments.find((e) => e.id === params.id);
	if (!thisEnrollment) return;
	return { loading: false, error: null, data: thisEnrollment };
};

export const mockApiOrganizationsOrgIdSitesSiteIdEnrollmentsIdDelete = () => (params: any) => ({
	loading: false,
	error: null,
});

export const mockApiOrganizationsOrgIdChildrenGet = (child: Child) => (params: any) => {
	const mappedChildToEnrollment = [child].reduce<{ [x: string]: Enrollment[] }>((acc, c) => {
		acc[c.id] = c.enrollments || [];
		return acc;
	}, {});
	return { loading: false, error: null, data: mappedChildToEnrollment };
};

export const mockApiOrganizationsOrgIdSitesIdGet = (site: Site) => (params: any) => ({
	loading: false,
	error: null,
	data: site,
});

export const mockApiOrganizationsOrgIdSitesSiteIdEnrollmentsGet = (enrollments: Enrollment[]) => (
	params: any
) => ({
	loading: false,
	error: null,
	data: enrollments.filter((e) => {
		return (
			(!e.entry ? true : moment(e.entry).isBefore(params.endDate)) &&
			(!e.exit ? true : moment(e.exit).isAfter(moment(params.startDate)))
		);
	}),
});

export const mockApiOrganizationsOrgIdReportsGet = (reports: CdcReport[]) => (params: any) => ({
	loading: false,
	error: null,
	data: reports,
});

export const mockApiOrganizationsIdGet = (organization: Organization) => (params: any) => ({
	loading: false,
	error: null,
	data: organization,
});

export const mockApiOrganizationsOrgIdReportsIdGet = (report: CdcReport) => (params: any) => ({
	loading: false,
	error: null,
	data: report,
});

/**
 * Mock implementation of the useApi hook that does not make a network request.
 *
 * @deprecated Please mock constructApi and parseError directly instead of using
 * this for future tests. See ChildInfo.tests.tsx for more information on how.
 *
 * If you remove this method, please update the comment in ChildInfo.tests.tsx
 * accordingly.
 */
export default (mockApi: any) => (
	query: (api: any, opt?: any) => any,
	opts: ApiParamOpts<any> = { skip: false }
) => {
	const { skip, callback, deps } = opts || {};

	useEffect(() => {
		if (callback && !skip) callback(null);
	}, [skip, ...(deps || [])]);

	return skip ? { error: null, data: null, loading: false } : query(mockApi);
};
