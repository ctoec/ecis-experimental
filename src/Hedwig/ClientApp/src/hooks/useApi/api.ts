import { HedwigApi, Configuration } from '../../generated';
import getCurrentHost from '../../utils/getCurrentHost';

export type Api = HedwigApi;

export const constructApi: (accessToken: string | null) => Api | null = (
	accessToken: string | null
) => {
	if (!accessToken) return null;

	return new HedwigApi(
		new Configuration({
			basePath: getCurrentHost(),
			apiKey: `Bearer ${accessToken}`,
		})
	);
};
