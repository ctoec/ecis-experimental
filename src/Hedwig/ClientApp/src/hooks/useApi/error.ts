import {
	ValidationProblemDetailsFromJSON,
	ProblemDetailsFromJSON,
	ValidationProblemDetails,
	ProblemDetails,
} from '../../generated';

export type ApiError = ValidationProblemDetails | ProblemDetails;

export const parseError: (error: any) => Promise<ApiError | null> = async (error: any) => {
	try {
		const jsonResponse = await error.json();
		if (error.status === 400) {
			return ValidationProblemDetailsFromJSON(jsonResponse) || null;
		} else {
			return ProblemDetailsFromJSON(jsonResponse) || null;
		}
	} catch (e) {
		console.error('Error cannot be converted to JSON');
		console.error(e);
		return ProblemDetailsFromJSON({
			detail: 'Inspect console for error',
			title: 'Unknown error',
		});
	}
};
