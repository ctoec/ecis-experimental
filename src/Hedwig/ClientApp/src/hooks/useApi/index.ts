export type { ApiError } from './error';
export type { ApiParamOpts } from './query';

export { paginate } from './paginate';
export { default } from './query';
