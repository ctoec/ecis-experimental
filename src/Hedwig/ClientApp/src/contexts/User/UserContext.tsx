import React from 'react';
import useApi from '../../hooks/useApi';
import { User } from '../../generated/models/User';

export type UserContextType = {
	user?: User;
	loading: boolean;
};

const UserContext = React.createContext<UserContextType>({
	user: undefined,
	loading: true,
});

const { Provider, Consumer } = UserContext;

export type UserProviderPropsType = {};

/**
 * Context Provider for injecting User object into React hierarchy
 *
 * @param props Props with user
 */
const UserProvider: React.FC<UserProviderPropsType> = ({ children }) => {
	const { loading, data: user } = useApi((api) => api.apiUsersCurrentGet());
	return <Provider value={{ loading, user }}>{children}</Provider>;
};

export { UserProvider };
export { Consumer as UserConsumer };
export default UserContext;
