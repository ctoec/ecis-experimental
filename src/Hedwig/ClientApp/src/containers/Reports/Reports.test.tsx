import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import 'react-dates/initialize';
import Reports from './Reports';
import UserContext from '../../contexts/User/UserContext';

const user = {
  id: 1,
  firstName: 'Minerva',
  lastName: 'McGonagall',
  orgPermissions: [
    {
      organizationId: 1,
      organization: {
        id: 1,
        name: "Children's Adventure Center",
        sites: [{ id: 1, name: "Children's Adventure Center", organizationId: 1 }],
      },
      id: 1,
      userId: 1,
    },
  ],
  sitePermissions: [],
};

// https://github.com/facebook/jest/issues/5579
jest.mock('../../hooks/useApi', () => {
  const _moment = require('moment');
  const moment = (_: any) => {
    return {
      ..._moment(_),
      getMonth: () => _moment(_).month()
    }
  }
  return {
    __esModule: true,
    default: (callback: any, dependencies: any[]) => {
      return callback(
        {
          apiOrganizationsIdGet: (params: any) => [
            false,
            null,
            {
              id: 1,
              name: "Children's Adventure Center",
              organizationId: 1,
              enrollments: undefined,
              organization: undefined,
            },
          ],
          apiOrganizationsOrgIdReportsGet: (params: any) => [
            false,
            null,
            [
              {
                id: 1,
                type: 'CDC',
                reportingPeriod: {
                  period: moment('2019-08-01'),
                  periodStart: moment('2019-07-29'),
                  periodEnd: moment('2019-09-01'),
                  dueAt: moment('2019-09-15'),
                },
                submittedAt: '2019-09-09',
                organization: {
                  id: 1,
                  name: "Children's Adventure Center",
                },
              },
              {
                id: 2,
                type: 'CDC',
                reportingPeriod: {
                  period: moment('2019-09-01'),
                  periodStart: moment('2019-09-02'),
                  periodEnd: moment('2019-09-29'),
                  dueAt: moment('2019-10-15'),
                },
                submittedAt: null,
                organization: {
                  id: 1,
                  name: "Children's Adventure Center",
                },
              },
            ],
          ],
        },
        []
      );
    },
  };
});

import useApi from '../../hooks/useApi';

afterAll(() => {
  jest.resetModules();
});

describe('Roster', () => {
  it('matches snapshot', () => {
    const wrapper = mount(
      <UserContext.Provider value={{ user }}>
        <BrowserRouter>
          <Reports />
        </BrowserRouter>
      </UserContext.Provider>
    );
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });
});