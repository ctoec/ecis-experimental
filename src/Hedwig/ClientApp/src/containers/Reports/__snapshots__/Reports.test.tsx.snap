// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster matches snapshot 1`] = `"<div class=\\"Reports\\"><section class=\\"grid-container\\"><h1>Reports</h1><table id=\\"reports-table\\" class=\\"oec-table oec-table--full-width\\"><thead><tr><th scope=\\"col\\" class=\\"oec-table__column-header oec-sortable\\"><span class=\\"oec-table__column-title\\">Period</span><div class=\\"oec-table__sort-controls\\"><input id=\\"reports-table-col0-asc\\" name=\\"reports-table-column-0-sort\\" type=\\"checkbox\\" value=\\"asc\\"><label class=\\"oec-table__sort-controls__asc\\" for=\\"reports-table-col0-asc\\"><span>Sort ascending</span></label><input id=\\"reports-table-col0-desc\\" name=\\"reports-table-column-0-sort\\" type=\\"checkbox\\" value=\\"desc\\" checked=\\"\\"><label class=\\"oec-table__sort-controls__desc\\" for=\\"reports-table-col0-desc\\"><span>Sort descending</span></label></div></th><th scope=\\"col\\" class=\\"oec-table__column-header oec-sortable\\"><span class=\\"oec-table__column-title\\">Type</span><div class=\\"oec-table__sort-controls\\"><input id=\\"reports-table-col1-asc\\" name=\\"reports-table-column-1-sort\\" type=\\"checkbox\\" value=\\"asc\\"><label class=\\"oec-table__sort-controls__asc\\" for=\\"reports-table-col1-asc\\"><span>Sort ascending</span></label><input id=\\"reports-table-col1-desc\\" name=\\"reports-table-column-1-sort\\" type=\\"checkbox\\" value=\\"desc\\"><label class=\\"oec-table__sort-controls__desc\\" for=\\"reports-table-col1-desc\\"><span>Sort descending</span></label></div></th><th scope=\\"col\\" class=\\"oec-table__column-header oec-sortable\\"><span class=\\"oec-table__column-title\\">Program/Site</span><div class=\\"oec-table__sort-controls\\"><input id=\\"reports-table-col2-asc\\" name=\\"reports-table-column-2-sort\\" type=\\"checkbox\\" value=\\"asc\\"><label class=\\"oec-table__sort-controls__asc\\" for=\\"reports-table-col2-asc\\"><span>Sort ascending</span></label><input id=\\"reports-table-col2-desc\\" name=\\"reports-table-column-2-sort\\" type=\\"checkbox\\" value=\\"desc\\"><label class=\\"oec-table__sort-controls__desc\\" for=\\"reports-table-col2-desc\\"><span>Sort descending</span></label></div></th><th scope=\\"col\\" class=\\"oec-table__column-header oec-sortable\\"><span class=\\"oec-table__column-title\\">Date submitted</span><div class=\\"oec-table__sort-controls\\"><input id=\\"reports-table-col3-asc\\" name=\\"reports-table-column-3-sort\\" type=\\"checkbox\\" value=\\"asc\\"><label class=\\"oec-table__sort-controls__asc\\" for=\\"reports-table-col3-asc\\"><span>Sort ascending</span></label><input id=\\"reports-table-col3-desc\\" name=\\"reports-table-column-3-sort\\" type=\\"checkbox\\" value=\\"desc\\"><label class=\\"oec-table__sort-controls__desc\\" for=\\"reports-table-col3-desc\\"><span>Sort descending</span></label></div></th></tr></thead><tbody><tr><th scope=\\"row\\"><a class=\\"usa-link\\" href=\\"/reports/2\\">September 2019</a>&nbsp;<span class=\\"oec-inline-icon oec-inline-icon--attentionNeeded\\"><svg>error.svg</svg><span class=\\"usa-sr-only\\">(attention needed)</span></span></th><td>CDC</td><td>Children's Adventure Center</td><td class=\\"oec-table__cell--gray\\">due 10/15/2019</td></tr><tr><th scope=\\"row\\"><a class=\\"usa-link\\" href=\\"/reports/1\\">August 2019</a>&nbsp;</th><td>CDC</td><td>Children's Adventure Center</td><td class=\\"oec-table__cell--tabular-nums\\">09/09/2019</td></tr></tbody></table></section></div>"`;
